languages/csharp/exercises/concept/constants/.docs/instructions.md

```diff
@@ -0,0 +1,21 @@
+The authentication system that you last saw in (TODO cross-ref-tba) is in need of some attention. You have been tasked with cleaning up the code.
+
+## 1. Set appropriate fields and properties to const
+
+This is a refactoring task. Add the `const` modifier to any members of `Authenticator` or `Identity` that you think appropriate.
+
+## 2. Set appropriate fields to readonly
+
+This is a refactoring task. Add the `readonly` modifier to any fields of the `Authenticator` class or the `Identity` struct that you think appropriate.
+
+## 3. Remove set accessor or make it private for any appropriate field
+
+Remove the `set` accessor or make it `private` for any appropriate property on the `Authenticator` class or `Identity` struct.
+
+## 4. Ensure that the admin cannot be tampered with
+
+At present the admin identity field is returned by a call to `GetAdmin()`. This is not ideal as the caller can modify the field. Find a way to prevent the caller modifying the details of admin on the `Authenticator` object.
+
+## 5. Ensure that the developers cannot be tampered with
+
+At present the dictionary containing the hard coded privileged developer identities is returned by a call to `GetDevelopers()`. This is not ideal as the caller can modify the dictionary. Find a way to prevent the caller modifying the details of admin on the `Authenticator` object.
```

Comment:
Will students know about `ReadOnlyDictionary`? If not, this might be something that we could introduce in this exercise too to make the exercise a bit more interesting.

Reply:
Done